src/tools/miri/src/provenance_gc.rs

```diff
@@ -196,6 +196,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriInterpCxExt<'mir, 'tcx> {
         let this = self.eval_context_mut();
         let allocs = LiveAllocs { ecx: this, collected: allocs };
         this.machine.allocation_spans.borrow_mut().retain(|id, _| allocs.is_live(*id));
+        this.machine.symbolic_alignment.borrow_mut().retain(|id, _| allocs.is_live(*id));
```

Comment:
This `is_live` check is the only reason that this hash map needs to be in a `RefCell` :/

Reply:
why does this need interior mutability at all? we have mutable access during allocation and here

Reply:
`allocs` borrows `this`, so we can't mutably borrow `this.machine` while `allocs` is alive.